Add code style rule for multiple return types (#27452)
element-hq/element-web: code_style.md

    }
    ```

37. Avoid functions whose behaviour / return type varies with different parameter types.

>     Multiple return types are fine when appropriate (eg. SDKConfig.get() with a string param which
>     returns the type according to the param given) but SDKConfig.get() with no args returning the
>     whole config object is not: this could just be a separate function.

Reviewer: It would be nice to define "when appropriate" a bit better.

```suggestion

    Multiple return types are acceptable for operations such as a lookup based on a string parameter
    `SDKConfig.get()`, whose return type varies according to the `key` parameter, is an example of this.
    
    However, completely changing the behaviour of the function based on the parameter types is not
    acceptable: instead, use a separate function. For example, it would not be acceptable for
    `SDKConfig.get()` with no arguments return the whole configuration object.
```

Author: Better hopefully?

    }
    ```

> 37. Avoid functions whose behaviour / return type varies with different parameter types.
>     Multiple return types are fine, but the function's fundamental behaviour should not
>     change according its arguments: have two separate functions. For example,SDKConfig.get()

Reviewer: I still think it's a bit weird that we say "Avoid functions whose ... return type varies" on one line and then "Multiple return types are fine" on the next line.

Maybe something like:

```suggestion
37. Avoid functions whose fundamental behaviour varies with different parameter types.
    Multiple return types are fine, but if the function's behaviour is going to change significantly,
    have two separate functions. For example, `SDKConfig.get()`
```

YMMV though. The whole thing is necessarily a bit woolly and no phrasing is going to be perfect.

Author: Yeah, probably better: applied manually
